<a href="https://colab.research.google.com/github/sandeepstele/Python-Codeforces-under-800/blob/main/MonteCarlo.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

 A card is drawn from a well-shuffled pack of 52 cards. Find the probability of getting:

‘2’ of spades


i)Initialize a variable called "success_count" to 0. This variable will keep track of the number of times we draw the '2' of spades.

Set the number of iterations for the Monte Carlo simulation. This determines how many times we will simulate drawing a card from the deck. Let's say we choose 1,000,000 iterations for a reasonably accurate result.

Run the following steps for the specified number of iterations:

a. Shuffle the deck of 52 cards randomly.

b. Draw the top card from the deck.

c. Check if the drawn card is the '2' of spades. If it is, increment the "success_count" variable by 1.

After completing all the iterations, calculate the probability by dividing the "success_count" by the total number of iterations (1,000,000 in our case).

In [2]:
import random

def monte_carlo_probability(iterations):
    success_count = 0
    deck = ['2S', '3S', '4S', '5S', '6S', '7S', '8S', '9S', '10S', 'JS', 'QS', 'KS', 'AS',
            '2H', '3H', '4H', '5H', '6H', '7H', '8H', '9H', '10H', 'JH', 'QH', 'KH', 'AH',
            '2D', '3D', '4D', '5D', '6D', '7D', '8D', '9D', '10D', 'JD', 'QD', 'KD', 'AD',
            '2C', '3C', '4C', '5C', '6C', '7C', '8C', '9C', '10C', 'JC', 'QC', 'KC', 'AC']

    for _ in range(iterations):
        random.shuffle(deck)
        drawn_card = deck[0]

        if drawn_card == '2S':
            success_count += 1

    probability = success_count / iterations
    return probability

# Run the Monte Carlo simulation with 1,000,000 iterations
iterations = 1000000
probability = monte_carlo_probability(iterations)

print("Probability of drawing the '2' of spades:", probability)

Probability of drawing the '2' of spades: 0.019458


Lets verify this practically using calculations!
Total no. of cards = 52

So total no. of possible outcomes, n(S) = 52

(i) Let E1 denotes the event of getting 2 of spades.

No. of ‘2’ of spades = 1

n(E1) = 1

P (getting 2 of spades) = no. of favourable outcomes/total no. of possible outcomes of E

= n(E1)/ n(S)

= 1/52
=0.0192307692307692


we see that 0.1923 is almost equal to 0.019458 found by the Monte Carlo method.